docs/web3-eth.rst

```diff
@@ -795,7 +795,7 @@ Returns a block matching the block number or block hash.
 Parameters
 ----------
 
-1. ``String|Number`` - The block number or block hash. Or the string ``"genesis"``, ``"latest"`` or ``"pending"`` as in the :ref:`default block parameter <eth-defaultblock>`.
+1. ``String|Number|BN`` - The block number or block hash. Or the string ``"genesis"``, ``"latest"`` or ``"pending"`` as in the :ref:`default block parameter <eth-defaultblock>`.
```

Comment:
```suggestion
1. ``String|Number|BN`` - The block number or block hash. Or the string ``"earliest"``, ``"latest"`` or ``"pending"`` as in the :ref:`default block parameter <eth-defaultblock>`.
```

Reply:
@nivida There's a few of these - does that change seem correct? 

Checked the codebase and I don't see "genesis" anywhere except the docs.

[JSON-RPC defaultBlock spec](https://github.com/ethereum/wiki/wiki/JSON-RPC#the-default-block-parameter)

Reply:
> @nivida There's a few of these - does that change seem correct?

Yes, this is correct. It was always possible but not documented in the developer documentation or the function docs of those ``formatters``.

> Checked the codebase and I don't see "genesis" anywhere except the docs.

``genesis`` is handled in the ``inputDefaultBlockNumberFormatter`` of the ``web3-core-helpers`` package.

Reply:
I've updated the related documentation and added a sentence about the pre-defined block numbers if required. The missing and compatible type ``BigNumber`` got added to the ``BlockNumber`` type of the web3-core module.

Reply:
Oh interesting! Thanks for double-checking that.